src/anemoi/models/preprocessing/imputer.py

```diff
@@ -109,11 +112,8 @@ def transform(self, x: torch.Tensor, in_place: bool = True) -> torch.Tensor:
         if not in_place:
             x = x.clone()
 
-        # Initilialize mask once
-        if self.nan_locations is None:
-            # The mask is only saved for the last two dimensions (grid, variable)
-            idx = [slice(0, 1)] * (x.ndim - 2) + [slice(None), slice(None)]
-            self.nan_locations = torch.isnan(x[idx].squeeze())
+        # Initilialize mask every time
+        self.nan_locations = torch.isnan(x)
```

Comment:
This change likely breaks for different batch sizes in training, validation, and inference. What is the reasoning behind removing the slicing?